src/applications/mhv-medical-records/tests/e2e/pages/VitalsListPage.js

```diff
@@ -29,19 +29,24 @@ class VitalsListPage extends BaseListPage {
     );
     cy.intercept('POST', '/v0/datadog_action', {}).as('datadogAction');
     cy.visit('my-health/medical-records/vitals');
-    cy.wait('@vitalsList');
-    cy.wait('@vamcEhr');
-    cy.wait('@mockUser');
-    cy.wait('@featureToggles');
-    cy.wait('@maintenanceWindow');
-    cy.wait('@status');
+    cy.wait([
+      '@vitalsList',
+      '@vamcEhr',
+      '@mockUser',
+      '@featureToggles',
+      '@maintenanceWindow',
+      '@status',
+    ]);
   };
 
-  clickLinkByRecordListItemIndex = (index = 0) => {
-    cy.findAllByTestId('record-list-item')
-      .find('a')
-      .eq(index)
-      .click();
+  clickLinkByRecordListItem = vitalsHeading => {
+    cy.contains(vitalsHeading, { includeShadowDom: true }).then(element => {
```

Comment:
Note: Key here is using the text instead of the `data-test-id`. This allows Cypress to retry finding the text and letting the page render. Before, the page was rendering at times with the `record-list-item` hidden and empty which created the flakiness.